test/gtest/core/test_alltoall.cc

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include "common/test_ucc.h"
+
+// tmp
+#ifdef ucc_max
+#undef ucc_max
+#endif
+
+#include "src/utils/ucc_math.h"
+
+using Param_0 = std::tuple<int, int, int>;
+using Param_1 = std::tuple<int, int>;
+
+class test_alltoall : public UccCollArgs, public ucc::test
+{
+private:
+    void init_proc_buf(int nprocs, int rank, uint8_t *buf, size_t len)
+    {
+        for (auto i = 0; i < len * nprocs; i++) {
+            buf[i] = (uint8_t)rank;
```

Comment:
In general i would suggest to initialize buffer with value that changes along the buffer. E.g.:
buf[i] = (uint8_t)((rank + i + 1) % UINT8_MAX).
This may not affect exactly pairwise alltoall, but for other algorithms this is more stressful test because checks that all bytes within a buffer are not mixed/broken